main/src/hadd.cxx

```diff
@@ -155,29 +160,62 @@
 #include <optional>
 #include <sstream>
 #include <string>
+#include <streambuf>
 
 #ifndef R__WIN32
 #include "ROOT/TProcessExecutor.hxx"
 #endif
 
 ////////////////////////////////////////////////////////////////////////////////
 
-inline std::ostream &Err()
+// NOTE: TFileMerger will use PrintLevel = gVerbosity - 1. If PrintLevel is < 1, it will print nothing, otherwise
+// it will print everything. To give some granularity to hadd, we do the following:
+// gVerbosity = 0: only print hadd errors
+// gVerbosity = 1: only print hadd errors + warnings
+// gVerbosity = 2: print hadd errors + warnings and TFileMerger messages
+// gVerbosity > 2: print all hadd and TFileMerger messages.
+static int gVerbosity = 99;
```

Comment:
Minor
```suggestion
static int gHaddVerbosity = 99;
```
to clarify to the reader (that is not also checking the declaration) of the point of use that this is limited to hadd.